packages/cloudflare/src/api/memory-queue.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { generateMessageGroupId } from "@opennextjs/aws/core/routing/queue.js";
+import { afterEach, beforeAll, describe, expect, it, vi } from "vitest";
+
+import cache, { DEFAULT_REVALIDATION_TIMEOUT } from "./memory-queue";
+
+vi.mock("./.next/prerender-manifest.json", () => Promise.resolve({ preview: { previewModeId: "id" } }));
+
+const defaultOpts = {
+  MessageBody: { host: "test.local", url: "/test" },
+  MessageGroupId: generateMessageGroupId("/test"),
+  MessageDeduplicationId: "",
+};
+
+describe("MemoryQueue", () => {
+  beforeAll(() => {
+    vi.useFakeTimers();
+    globalThis.internalFetch = vi.fn().mockReturnValue(new Promise((res) => setTimeout(() => res(true), 1)));
+  });
+
+  afterEach(() => vi.clearAllMocks());
+
+  it("should process revalidations for a path", async () => {
+    const firstRequest = cache.send(defaultOpts);
+    vi.advanceTimersByTime(DEFAULT_REVALIDATION_TIMEOUT);
+    await firstRequest;
```

Comment:
Wondering if that would make more sense?

```suggestion
    await cache.send(defaultOpts);
    vi.advanceTimersByTime(DEFAULT_REVALIDATION_TIMEOUT);
```

Reply:
this would cause the test to time out as it would be awaiting on the fetch, and neither the fetch or the settimeout would resolve because we haven't advanced the timers.